app/controllers/characters_controller.rb

```diff
@@ -69,6 +69,19 @@ def update
     end
   end
 
+  def duplicate
+    @dup = @character.dup
```

Comment:
How would I catch a transaction failing? (So I can redirect them with an error.) If that's not the point of a transaction… what is? (If an error occurs, it'll bubble up and give them a 500, and then it'll also end up rolling back all the changes?)